python/pipeline/pupil.py

```diff
@@ -540,4 +575,463 @@ def make(self, key):
+        config['config_path'] = temp_config['config_path']
+        config['shuffle'] = temp_config['shuffle']
+        config['trainingsetindex'] = temp_config['trainingsetindex']
+
+        trainFraction = config['TrainingFraction'][config['trainingsetindex']]
+        DLCscorer = auxiliaryfunctions.GetScorerName(
+            config, config['shuffle'], trainFraction)
+
+        # make needed directories
+        tracking_dir, original_video_path = self.create_tracking_directory(key)
+        self.insert1(
+            dict(key, config_path=config['config_path'], tracking_dir=tracking_dir))
+
+        # make a short video (5 seconds long)
+        short_video_path, original_width, original_height, mid_frame_num = self.make_short_video(
+            tracking_dir)
+
+        # save info about original video
+        original_video = self.OriginalVideo()
+        original_video.insert1(
+            dict(key, original_width=original_width,
+                 original_height=original_height,
+                 video_path=original_video_path))
+
+        # save info about short video
+        short_video = self.ShortVideo()
+        short_video.insert1(
+            dict(key, starting_frame=mid_frame_num, video_path=short_video_path))
+
+        short_h5_path = short_video_path.split('.')[0] + DLCscorer + '.h5'
+
+        # predict using the short video
+        self.predict_labels(short_video_path, config)
+
+        # obtain the cropping coordinates from the prediciton on short video
+        cropped_coords = self.obtain_cropping_coords(
+            short_h5_path, DLCscorer, config)
+
+        # add 100 pixels around cropping coords. Ensure that it is within the original dim
+        pixel_num = 100
+        cropped_coords = self.add_pixels(cropped_coords=cropped_coords,
+                                         original_width=original_width,
+                                         original_height=original_height,
+                                         pixel_num=pixel_num)
+
+        # make a compressed and cropped video
+        compressed_cropped_video_path = self.make_compressed_cropped_video(
+            tracking_dir, cropped_coords)
+
+        # predict using the compressed and cropped video
+        self.predict_labels(compressed_cropped_video_path, config)
+
+        compressed_cropped_video = self.CompressedCroppedVideo()
+        compressed_cropped_video.insert1(dict(key, cropped_x0=cropped_coords['cropped_x0'],
+                                              cropped_x1=cropped_coords['cropped_x1'],
+                                              cropped_y0=cropped_coords['cropped_y0'],
+                                              cropped_y1=cropped_coords['cropped_y1'],
+                                              added_pixels=pixel_num,
+                                              video_path=compressed_cropped_video_path))
+
+
+@schema
+class FittedContourDeeplabcut(dj.Computed):
```

Comment:
Sure we want this to be a different table rather than a different method on whatever table we already used? Did you check with somebody about that? They'll both require changes in downstream code but not sure which is the best way to go about it